elastic/kibana
# [ResponseOps][Alerts] Implement platform alerts grouping components

## Summary

Adds solution-agnostic components to create hierarchical alerts grouping UIs, adapting the original implementation from Security Solution.

Closes #184398 

## To Verify

For existing usages of the `@kbn/grouping` package: verify that the grouped UIs work correctly (Security Alerts, Cloud Security Posture).

New alerting UI components: checkout https://github.com/elastic/kibana/pull/183114 (PoC PR), where the updated `@kbn/grouping` package and these new components are used in Observability's main Alerts page.

### Checklist
